src/main/scala/com.snowplowanalytics.dataflow/streaming/StreamingCounts.scala

```diff
@@ -0,0 +1,166 @@
+import com.google.cloud.dataflow.sdk.transforms.ParDo
+import com.google.cloud.dataflow.sdk.transforms.PTransform
+import com.google.cloud.dataflow.sdk.transforms.GroupByKey
+import com.google.cloud.dataflow.sdk.transforms.windowing.FixedWindows
+import com.google.cloud.dataflow.sdk.transforms.windowing.Window
+import com.google.cloud.dataflow.sdk.values.KV
+import com.google.cloud.dataflow.sdk.values.PDone
+import com.google.cloud.dataflow.sdk.values.PCollection
+
+// Bigtable
+import com.google.cloud.bigtable.dataflow.AbstractCloudBigtableTableDoFn
+import com.google.cloud.bigtable.dataflow.CloudBigtableTableConfiguration
+import com.google.cloud.bigtable.dataflow.CloudBigtableTableConfiguration.Builder
+
+//HBase
+import org.apache.hadoop.hbase.client.Connection
+
+// This project
+import storage.BigtableUtils
+
+object StreamingCounts {
+
+  /**
+   * Private function to set up Dataflow and Bigtable
+   *
+   * @param config The configuration for our job using StreamingCountsConfig.scala
+   */
+  private def setupDataflowAndBigtable(config: StreamingCountsConfig): (Pipeline, CloudBigtableTableConfiguration) = {
+    PipelineOptionsFactory.register(classOf[DataflowPipelineOptions])
+
+    val options = PipelineOptionsFactory
+      .as(classOf[DataflowPipelineOptions])
+
+    options.setRunner(classOf[DataflowPipelineRunner])
+    options.setProject(config.projectId)
+    options.setStagingLocation(config.stagingLocation)
+    options.setStreaming(true)
+
+    val bigtableConfig = new CloudBigtableTableConfiguration.Builder()
+        .withProjectId(config.projectId)
+        .withInstanceId(config.instanceId)
+        .withTableId(config.tableName)
+        .build()
+
+    val pipeline = Pipeline.create(options)
+
+    (pipeline, bigtableConfig)
+  }
+
+  /**
+   * function applied in the "map" phase
+   */
+  def bucketEvents = new DoFn[String,KV[String,String]]() {
+
+	//val LOG = Logger.getLogger("bucketEvents")
+
+    override def processElement(c: DoFn[String,KV[String,String]]#ProcessContext) {
+        val e = SimpleEvent.fromJson(c.element)
+        c.output(KV.of(e.bucket, e.`type`))
+    }
+  }
+
+  /**
+   * transform applied to count the events per type in each bucket and 
+   * store the counts in Bigtable
+   */
+
+  class StoreEventCounts(config: CloudBigtableTableConfiguration, tableName: String)
+            extends PTransform[PCollection[KV[String, JIterable[String]]], PDone] {
+
+    def storeEventCounts(config: CloudBigtableTableConfiguration, tableName: String) = new AbstractCloudBigtableTableDoFn[KV[String,JIterable[String]], String](config) {
+
+      override def processElement(c: DoFn[KV[String,JIterable[String]], String]#ProcessContext) {
+          val bucketName = c.element.getKey
+          
+          val counts = c.element.getValue.asScala.groupBy(identity).mapValues(_.size)
+          for ((eventType, count) <- counts) {
+              BigtableUtils.setOrUpdateCount(
+                  getConnection,
```

Comment:
You might want to comment that `getConnection` is provided by `AbstractCloudBigtableTableDoFn`